resources/js/bots/wallet/bot.js

```diff
@@ -403,25 +402,29 @@ function handleSend(params, context) {
 
     web3.eth.sendTransaction(data, function(error, hash) {
         if (error) {
-            status.sendSignal("handler-result", {
-                status: "failed",
+            /*status.sendSignal("handler-result", {
                 error: {
                     markup: status.components.validationMessage(
                         I18n.t('validation_tx_title'),
                         I18n.t('validation_tx_failed')
                     )
                 },
                 origParams: context["orig-params"]
-            });
+            });*/
         } else {
             status.sendSignal("handler-result", {
-                status: "success",
-                hash: hash,
+                result: {
+                    hash: hash
+                },
                 origParams: context["orig-params"]
             });
         } 
     });
-    // async handler, so we don't return anything immediately
+    
+    // jail async handler, we have to signal it
+    return {
+        flow: "jailAsync"
```

Comment:
what does this mean?

Reply:
the same for "appAsync"

Reply:
This signalises the workflow which should be taken when evaluating result of the command handler. There will be actually 3 different possibilities:
* Normal synchronous handler, where in the app, upon receiving the result, command-message processing immediately resumes, nothing changes there, we just return data.
* Jail asynchronous handler(the case here), where we return map with `flow` property set to `"jailAsync"` to signalise that processing of command-message should stop and resume only upon signal from jail (`"jail.signal"` of type `"handler-result"`)
* App asynchronous handler (the one I originally planned to use but didn't in the end) where besides flow, you provide `hook` and `data` properties in the return map, this will signalise that you want the processing to go async, but signal to resume won't be from jail, but some module from the RN app (like wallet, contacts, camera capture, etc.).
In the future, we should create and describe such application hooks, which anybody can use when writing their custom commands for status, that way, we could get rid of lot of hardcoded functionality tied to particular commands like it's now.